projects/demo/src/pages/documentation/js-frameworks-angular/examples/1-nested/component.ts

```diff
@@ -15,5 +15,5 @@ export class NestedDocExample1 {
     };
 
     readonly predicate: MaskitoElementPredicate = element =>
-        element.querySelector('tui-input input')!;
+        element.querySelector<HTMLInputElement>('tui-input input')!;
```

Comment:
I think this doesn't really do anything, because this arrow function is already typed as `MaskitoElementPredicate`, and narrowing it down from `HTMLInputElement | HTMLTextAreaElement` is irrelecant.

Reply:
I'll research this further, once the `Promise` is added to the `MaskitoElementPredicate` Typescript produces this error:
```
Type 'Element' is not assignable to type 'HTMLInputElement | HTMLTextAreaElement | Promise<HTMLInputElement | HTMLTextAreaElement>'.

Type 'Element' is missing the following properties from type 'HTMLTextAreaElement': autocomplete, cols, defaultValue, dirName, and 137 more.ts
```

This applies to: https://github.com/Tinkoff/maskito/pull/316#discussion_r1203563251 as well. 

This may be a weird quirk or bug in Typescript. I'm also observing that if the selector is updated to:

```ts
element.querySelector('input')!;
```

The error goes away and the inferred type is `HTMLInputElement`. With `tui-input input`, the inferred type is `Element`. 

If the `Promise` type is removed from `MaskitoElementPredicate`, using `tui-input input` results in the inferred type of `HTMLInputElement`.